.drone/drone.yml

```diff
@@ -122,6 +122,52 @@ steps:
 depends_on:
   - Test
 
+volumes:
+  - name: docker
+    host:
+      path: /var/run/docker.sock
+---
+kind: pipeline
+type: docker
+name: Deploy-To-Deployment-Tools 
+platform:
+  os: linux
+  arch: amd64
+trigger:
+  ref:
+    - refs/heads/main
+
+steps:
+  - name: Update Deployment Tools 
+    image: us.gcr.io/kubernetes-dev/drone/plugins/updater
+    settings:
+      config_json: |- 
+        {
+          "destination_branch": "master",
+          "pull_request_branch_prefix": "cd-agent",
+          "pull_request_enabled": true,
+          "pull_request_existing_strategy": "fail",
+          "pull_request_message": "Created by Drone build https://drone.grafana.net/api/repos/grafana/agent/builds/${DRONE_BUILD_NUMBER}\n\nBuild commit: ${DRONE_COMMIT_LINK}\n",
+          "pull_request_team_reviewers": [
+            "rfratto",
+            "mattdurham",
+            "gouthamve"
+          ],
+          "repo_name": "deployment_tools",
+          "update_jsonnet_attribute_configs": [
+            {
+              "file_path": "ksonnet/environments/platform-observability/waves/agent.libsonnet",
+              "jsonnet_key": "dev_canary",
+              "jsonnet_value": "grafana/agent:$DRONE_TAG"
```

Comment:
If I understand https://docs.drone.io/pipeline/environment/reference/drone-tag/ correctly ,`DRONE_TAG` will be empty for this pipeline for some of the events, see the trigger section:

```
trigger:
  ref:
    - refs/heads/main
```

You are not filtering on a specific event (such as: `push`, `pull_request` or `tag`) so it would apply to all fo them.

Reply:
Hrm, this is true. I need to somehow get `IMAGE_TAG ?= $(shell ./tools/image-tag)` in there. Not sure how. Looking at how other repos are doing it.

Reply:
I feel responsible for this being a pain so I opened grafana/deployment_tools#20048 to add `jsonnet_value_file` support. 

Once merged and the updated image gets deployed, we can use an initial stage to write the output of `$(shell ./tools/image-tag)` to a file and use the file name in the config_json instead.

Reply:
That PR has been merged and the docker image is updated to include the new feature.